src/Core/Repositories/ICollectionRepository.cs

```diff
@@ -13,6 +13,7 @@ public interface ICollectionRepository : IRepository<Collection, Guid>
     Task<ICollection<Tuple<Collection, CollectionAccessDetails>>> GetManyByUserIdWithAccessAsync(Guid userId, Guid organizationId);
     Task<CollectionDetails> GetByIdAsync(Guid id, Guid userId);
     Task<ICollection<Collection>> GetManyByManyIdsAsync(IEnumerable<Guid> collectionIds);
+    Task<ICollection<Collection>> GetManyByManyIdsAndOrgIdAsync(IEnumerable<Guid> collectionIds, Guid organizationId);
```

Comment:
What does this do for us beyond `GetMantByMantIdsAsync`?

Reply:
Gets Many folders from many Ids with that OrdId. Ensuring that you are trying to get collections from your org, if the collection id matches but the OrgId doesn't then should create a new one.

Reply:
my point was that this feels better handled by a logic check in the service vs pushing that logic into db